### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Thinus01**
### General requirements

- Made sure that there are no linter errors.
- Made sure that I used correct flow Gitflow.
- Made sure that I documented my work in a professional way.

## Project requirements

- Eliminate the current_user method from the ApplicationController since it will be supplied by devise.
- Install and configure devise.
- Enable users to register.
- Allow users to log in using their email and password. The hashed password should be saved in the database.
- Request email confirmation.
- Enable password resetting.
- Adjust the devise views for registration and login to match the Sneak Peek's wireframes and your design style.

### CODE REVIEW

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Thinus01 and @Mutai-Gilbert  , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have linter checks running, kindly ensure you have setup your linters config correctly and all linter checks are running and passing 
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (julie-ify) [inline app/views/devise/registrations/new.html.erb:33]: - [ ] Based on the wireframe layout, the form inputs should be centered on the page, but your form inputs are placed on the left. Kindly fix that.

  image

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (julie-ify) [inline app/views/devise/sessions/new.html.erb:26]: - [ ] Your login page inputs should be centered on the page, according to the wireframe. Kindly fix that.

  fkflfo

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Thinus01 and @Mutai-Gilbert,

Good job so far!

## Highlights!
- Good code organization and working devise features ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
```
Comment (ishimwezachee) [inline app/views/devise/registrations/new.html.erb:32]: - [ OPTIONAL] You did great to center the both sign-in and sign-up inputs field as previous reviewer suggested, but if you look very well on the wireframe., you can see that even the labels such as email, password, etc. Are placeholders, they are inside the input field. You can apply that, and center the rest of the text for better look as well.

**Review by ishimwezachee — APPROVED** [review]
Hi @Thinus01 and @Mutai-Gilbert,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image
## Highlights
- Good code organization and working devise features ✔️
- Descriptive README ✔️
- Implemented the requested changes from previous reviewer ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**